python/ray/data/_internal/execution/streaming_executor_state.py

```diff
@@ -548,14 +587,41 @@ def select_operator_to_run(
     ops = []
     for op, state in topology.items():
         if resource_manager.op_resource_allocator_enabled():
-            under_resource_limits = (
-                resource_manager.op_resource_allocator.can_submit_new_task(op)
+            can_submit_with_reason = (
+                resource_manager.op_resource_allocator.can_submit_new_task_with_reason(
+                    op
+                )
+            )
+            under_resource_limits = can_submit_with_reason["can_submit"]
+            limit_reason = (
+                f"(new tasks: {''.join(can_submit_with_reason['usage'])})"
+                if not under_resource_limits
+                else ""
             )
         else:
-            under_resource_limits = _execution_allowed(op, resource_manager)
-        in_backpressure = not under_resource_limits or any(
-            not p.can_add_input(op) for p in backpressure_policies
+            allowed_execution = _execution_allowed_with_reason(op, resource_manager)
+            under_resource_limits = allowed_execution["allowed_exec"]
+            limit_reason = (
+                f"(exec allowed: {''.join(allowed_execution['usage'])})"
+                if not under_resource_limits
+                else ""
+            )
+        in_backpressure = not under_resource_limits
+        in_backpressure_reason = (
+            f"under_resource_limits-> {limit_reason}" if in_backpressure else ""
         )
+        # backpressure policies
+        if not in_backpressure:
+            for p in backpressure_policies:
+                if not p.can_add_input(op):
+                    in_backpressure = True
+                    in_backpressure_reason = p.__class__.__name__
+                    break
+        # should add input
+        if not in_backpressure and not op.should_add_input():
+            in_backpressure = True
+            in_backpressure_reason = "op.should_add_input"
```

Comment:
We need to make the reasons easier to understand. E.g. `op.should_add_input` is unclear.